### PR TITLE
Ha improving training efficiency

### PR DESCRIPTION
**Pull request by hfadams**
Increased efficiency of training random forest model, fixed bug with pairing training rasters with ground truth, and added "tuning objective" user input.

## Summary by CodeRabbit

* New Features
  * Added configurable model tuning objective (Accuracy, Specificity, Sensitivity, Precision, Balanced, Youden) influencing automatic thresholding.
  * Expanded training/test sampling with per-timestep handling, spatial balance, optional edge enrichment, and detailed sampling info.
* Refactor
  * Overhauled Random Forest training with two-stage hyperparameter tuning and improved probability handling, enhancing model quality.
  * Updated evaluation to use explicit class labels and probabilities for Random Forest results.
* Chores
  * Bumped package version to 2.2.0.
  * Added “Model tuning objective” option to Advanced Classifier settings.

### CODE REVIEW

File: src/1-train-classifier.R [hunk 1]
```diff
@@ -68,6 +66,7 @@ setManualThreshold <- inputVariables[[7]]
 manualThreshold <- inputVariables[[8]]
 normalizeRasters <- inputVariables[[9]]
 rasterDecimalPlaces <- inputVariables[[10]]
+tuningObjective <- inputVariables[[11]]
```
[the remark this answers is not in the document]
Comment (hfadams) [reply src/1-train-classifier.R:69]: The tuningObjective object doesn't need to be validated since it is a user-supplied input that is limited to be selected from the list of possible options. 

The getRandomForestModel function uses probability = FALSE in the ranger calls for the model tuning steps and later set to probability = TRUE for the final model with the intention of increasing the efficiency of the tuning stages while maintaining necessary outputs from the final model. It isn't necessary to update the ranger arguments for the model tuning stages.

File: src/functions/0.2-training-and-prediction.r [hunk 2]
```diff
@@ -39,399 +39,424 @@ splitTrainTest <- function(
...
+  if (all(!is.finite(score_constrained))) {
+    warning("No threshold satisfies the provided constraints; returning unconstrained optimum.")
+    best_idx <- which.max(score)  # fallback
+  } else {
+    best_idx <- which.max(score_constrained)
+  }
 
-  return(optimalYouden)
+  thresholds[best_idx]
 }
```
[the remark this answers is not in the document]
Comment (hfadams) [reply src/functions/0.2-training-and-prediction.r:460]: Earlier in the data processing pipeline, presence data for the Random Forest model is converted to labeled factors ("presence" and "absence"), while the presence data for the CNN and MaxEnt models are not converted to factors. Since these data types are ensured earlier in the workflow it is not necessary to convert them to 0/1 here.

[the remark this answers is not in the document]
Comment (afilazzola) [reply src/functions/0.2-training-and-prediction.r:460]: Good

**Review by afilazzola — APPROVED** [review]
Overall changes look good and a big improvement
